pkg/api/v1alpha1/cluster.go

```diff
@@ -452,24 +453,43 @@ func validateCNIConfig(cniConfig *CNIConfig) error {
 		return fmt.Errorf("cni not specified")
 	}
 	var cniPluginSpecified int
+	var allErrs []error
 
 	if cniConfig.Cilium != nil {
 		cniPluginSpecified++
+		if err := validateCiliumConfig(cniConfig.Cilium); err != nil {
+			allErrs = append(allErrs, err)
+		}
 	}
 
 	if cniConfig.Kindnetd != nil {
 		cniPluginSpecified++
 	}
 
 	if cniPluginSpecified == 0 {
-		return fmt.Errorf("no cni plugin specified")
+		allErrs = append(allErrs, fmt.Errorf("no cni plugin specified"))
 	} else if cniPluginSpecified > 1 {
-		return fmt.Errorf("cannot specify more than one cni plugins")
+		allErrs = append(allErrs, fmt.Errorf("cannot specify more than one cni plugins"))
+	}
+
+	if len(allErrs) > 0 {
+		aggregate := utilerrors.NewAggregate(allErrs)
+		return fmt.Errorf("error validating cniConfig: %v", aggregate)
 	}
 
 	return nil
 }
 
+func validateCiliumConfig(cilium *CiliumConfig) error {
```

Comment:
instead of using type `string`, how about making it
```go
const (
         CiliumPolicyModeDefault CiliumPolicyEnforcementMode = "default"
         CiliumPolicyModeAlways CiliumPolicyEnforcementMode = "always"
         CiliumPolicyModeNever CiliumPolicyEnforcementMode = "never"
)
type CiliumConfig struct {
	// PolicyEnforcementMode determines communication allowed between pods. Accepted values are default, always, never.
	PolicyEnforcementMode CiliumPolicyEnforcementMode `json:"policyEnforcementMode,omitempty"`
}
```